eksctl/nasa-cryo.jsonnet

```diff
@@ -0,0 +1,109 @@
+local masterAzs = ["us-west-2a", "us-west-2b", "us-west-2c"];
+local nodeAz = "us-west-2a";
+
+// Node definitions for notebook nodes. Config here is merged
+// with our notebook node definition.
+// A `node.kubernetes.io/instance-type label is added, so pods
+// can request a particular kind of node with a nodeSelector
+local notebookNodes = [
+    { instanceType: "m5.large" },
+    { instanceType: "m5.xlarge" },
+    { instanceType: "m5.2xlarge" },
+    { instanceType: "m5.8xlarge" },
+];
+
+// Node definitions for dask worker nodes. Config here is merged
+// with our dask worker node definition, which uses spot instances.
+// A `node.kubernetes.io/instance-type label is set to the name of the
+// *first* item in instanceDistribution.instanceTypes, to match
+// what we do with notebook nodes. Pods can request a particular
+// kind of node with a nodeSelector
+local daskNodes = [
+    { instancesDistribution+: { instanceTypes: ["m5.large"] }},
+    { instancesDistribution+: { instanceTypes: ["m5.xlarge"] }},
+    { instancesDistribution+: { instanceTypes: ["m5.2xlarge"] }},
+    { instancesDistribution+: { instanceTypes: ["m5.8xlarge"] }},
+];
+
+{
+    apiVersion: 'eksctl.io/v1alpha5',
+    kind: 'ClusterConfig',
+    metadata+: {
+        name: "nasa-cryo",
+        region: clusterRegion,
+        version: '1.22'
+    },
+    availabilityZones: masterAzs,
+    iam: {
+        withOIDC: true,
+    },
+    nodeGroups: [
+        ng {
+            name: 'core-a',
+            availabilityZones: [nodeAz],
+            ssh: {
+                publicKeyPath: 'ssh-keys/nasa-cryo.key.pub'
+            },
+            instanceType: "m5.xlarge",
+            minSize: 1,
+            maxSize: 6,
+            labels+: {
+                "hub.jupyter.org/node-purpose": "core",
+                "k8s.dask.org/node-purpose": "core"
+            },
+        },
+    ] + [
+        ng {
+            // NodeGroup names can't have a '.' in them, while
+            // instanceTypes always have a .
+            name: "nb-%s" % std.strReplace(n.instanceType, ".", "-"),
+            availabilityZones: [nodeAz],
+            minSize: 0,
+            maxSize: 500,
+            instanceType: n.instanceType,
+            ssh: {
+                publicKeyPath: 'ssh-keys/nasa-cryo.key.pub'
+            },
+            labels+: {
+                "hub.jupyter.org/node-purpose": "user",
+                "k8s.dask.org/node-purpose": "scheduler"
+            },
+            taints+: {
+                "hub.jupyter.org_dedicated": "user:NoSchedule",
+                "hub.jupyter.org/dedicated": "user:NoSchedule"
+            },
+
+        } + n for n in notebookNodes
+    ] + [
+        ng {
+            // NodeGroup names can't have a '.' in them, while
+            // instanceTypes always have a .
+            name: "dask-%s" % std.strReplace(n.instancesDistribution.instanceTypes[0], ".", "-"),
+            availabilityZones: [nodeAz],
+            minSize: 0,
```

Comment:
Wondering what would happen if we would force the creation of one notebook node here. Would the server be allocated to it if it would exist? Or is it rather a node creation issue

Reply:
That worked!